File: 169-majority-element/majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {

        int majority = nums[0];
        int votes = 1;

        for(int i = 1; i < nums.Length; i++) {

            if(majority == nums[i]) {
                votes++;
            } else {
                votes--;
                if(votes == 0) {
                    votes = 1;
                    majority = nums[i];
                }
            }
        }
        return majority;

    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        majority = nums[0]
        votes = 1

        for i in range(1, len(nums)):
            if majority == nums[i]:
                votes += 1
            else:
                votes -= 1
                if votes == 0:
                    votes = 1
                    majority = nums[i]
        
        return majority